### ISSUE TITLE
Generator with unreachable yield is still a generator

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#453]
**Describe the bug**
In my latest attempt to use Python in the most unorthodox way imaginable, I found out that Pyright treats generators with unreachable yields as regular functions, reporting:

```json
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Expression of type 'None' cannot be assigned to return type 'Iterable[str]'\n  'None' cannot be assigned to 'Iterable[str]'",
	"source": "pyright",
	"startLineNumber": 5,
	"startColumn": 5,
	"endLineNumber": 5,
	"endColumn": 11
}
{
	"resource": "test.py",
	"owner": "_generated_diagnostic_collection_name_#0",
	"severity": 8,
	"message": "Expression of type 'None' cannot be assigned to return type 'AsyncIterable[str]'\n  'None' cannot be assigned to 'AsyncIterable[str]'",
	"source": "pyright",
	"startLineNumber": 10,
	"startColumn": 5,
	"endLineNumber": 10,
	"endColumn": 11
}
```

However, `foo` and `afoo` return a generator and an asynchronous generator, respectively, which can be generalised as an (asynchronous) iterator.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `foo` and `afoo` would be generators with a default return type of `None`.

**Screenshots or Code**
```py
from typing import Iterable, AsyncIterable


def foo() -> Iterable[str]:
    return
    yield ''


async def afoo() -> AsyncIterable[str]:
    return
    yield ''
```

**VS Code extension or command-line**
extension, 1.1.14

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now fixed in 1.1.15, which I just published.
